modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,398 @@
+      wdir_rad <- udunits2::ud.convert(ncdata_dir,"degrees","radians")
+      ncdata_e <- ncdata_spd * cos(wdir_rad)
+      ncdata_n <- ncdata_spd * sin(wdir_rad)
+      ncdf4::ncvar_put(nc, varid = Ewind.var, vals = ncdata_e)
+      ncdf4::ncvar_put(nc, varid = Nwind.var, vals = ncdata_n)
+    } else {
+      logger.warn("No NEON Wind data")
+    }
+    
+    # NEON.DP1.00041 Soil temp    (take 2cm level which is level 501)
+    soilTLoc = grep("DP1\\.00041",availProducts)
+    soilTDates <- neonmet.getDates(availDates,soilTLoc,startMon,endMon)
+    if (length(soilTDates>0)) {
+      if (verbose) {
+        logger.info("Reading NEON Soil Temp")
+      }
+      soilT.var <- ncdf4::ncvar_def(name = "soil_temperature", 
+                                  units = "K", 
+                                  dim = xytdim)
+      nc <- ncdf4::ncvar_add(nc = nc, v = soilT.var, verbose = verbose)
+      ncdata <- neonmet.getVals(dates=soilTDates,product=availProducts[soilTLoc[1]],site=site,
+                                datetime=datetime,data_col="soilTempMean",
+                                urlstring = "\\.00000\\.001\\.5..\\.(.*)_30_minute",
+                                units=c("celsius", "K"),belowground=TRUE)      
+      ncdf4::ncvar_put(nc, varid = soilT.var, vals = ncdata)
+      browser()
+    } else {
+      logger.warn("No NEON Soil Temp")
+    }
+    
+    # NEON.DP1.00034 CO2 at tower top (alt NEON.DP3.00009 CO2 profile) - not yet avail, don't have variable names
+  ncdf4::nc_close(nc)
+  } #For loop
+  return(invisible(results))
+} #function
+
+neonmet.getDates <- function(availDates,Loc,startMon,endMon) {
+  if (length(Loc)>0) { 
+    Dates <- unlist(availDates[Loc[1]])
+    GoodDates <- which((Dates >= startMon) & (Dates <= endMon))
+    if (length(GoodDates)>0) {
+      return(Dates[GoodDates])
+    } else {
+      return(NULL)
+    }
+  } else { 
+    return(NULL)
+  }  
+}
+
+neonmet.getVals <- function(dates,product,site,datetime,
+                            data_col,QF_col="finalQF", QF=0,
+                            urlstring = "\\.00000\\.000\\.(.*)30min",units,FillValue=NA,belowground=FALSE) {
+  ncdata <- rep(FillValue,length(datetime))
+  for (mon in dates) {
+    neonData <- nneo::nneo_data(product_code = product, site_code = site, year_month = mon)
+    urls <- neonData$data$urls
+    if (length(urls)>0) {
+      #Extract and read 30 minute data from the highest vertical level among files returned
+      #If belowground, then take top most level (lowest value)
+      if (belowground==TRUE) {
+        url30 <- head(sort(urls[grep(urlstring,urls)]),1)        
+      } else {
+        url30 <- tail(sort(urls[grep(urlstring,urls)]),1)        
+      }
+      if (length(url30)!=0) {
+        csvData <- read.csv(url30,stringsAsFactors=FALSE,header=TRUE)
+        #Retreive time dimension and figure out where in array to put it
+        csvDateTime <- as.POSIXct(gsub("T"," ",csvData$startDateTime),tz="UTC")
```

Comment:
No change required, but the `stringr` package from the Hadleyverse provides some nice substitutes for string matching functions, in case you have to do more complex string operations in the future.